java/ql/consistency-queries/calls.ql

```diff
@@ -1,5 +1,7 @@
 import java
 
 from MethodAccess ma
-where not exists(ma.getQualifier()) and ma.getFile().isKotlinSourceFile()
+// Generally Kotlin calls will always use an explicit qualifier, except for calls
+// to the synthetic instance initializer <obinit>, which use an implicit `this`.
+where not exists(ma.getQualifier()) and ma.getFile().isKotlinSourceFile() and not ma.getCallee() instanceof InstanceInitializer
```

Comment:
Should we add an explicit `this` qualifier to the extraction of `IrInstanceInitializerCall`?

Reply:
We could try; not sure how to make the right expression though so figured it was better to just rely on the QL lib's existing support for implicit `this`.

Reply:
I thought it would be simply like this: https://github.com/github/codeql/blob/eb1b3f801d080ec2f02e5c3bbbd2a697de0470f0/java/kotlin-extractor/src/main/kotlin/KotlinFileExtractor.kt#L2559-L2566